src/OrleansRuntime/Hosting/DefaultSiloServices.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Orleans.CodeGeneration;
+using Orleans.Configuration;
+using Orleans.GrainDirectory;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.ConsistentRing;
+using Orleans.Runtime.Counters;
+using Orleans.Runtime.GrainDirectory;
+using Orleans.Runtime.LogConsistency;
+using Orleans.Runtime.MembershipService;
+using Orleans.Runtime.Messaging;
+using Orleans.Runtime.MultiClusterNetwork;
+using Orleans.Runtime.Placement;
+using Orleans.Runtime.Providers;
+using Orleans.Runtime.ReminderService;
+using Orleans.Runtime.Scheduler;
+using Orleans.Runtime.Utilities;
+using Orleans.Runtime.Versions;
+using Orleans.Runtime.Versions.Compatibility;
+using Orleans.Runtime.Versions.Selector;
+using Orleans.Serialization;
+using Orleans.Streams;
+using Orleans.Streams.Core;
+using Orleans.Timers;
+using Orleans.Versions;
+using Orleans.Versions.Compatibility;
+using Orleans.Versions.Selector;
+
+namespace Orleans.Runtime.Hosting
+{
+    internal static class DefaultSiloServices
+    {
+        internal static void AddDefaultServices(IServiceCollection services)
+        {
+            services.AddOptions();
+
+            // Register system services.
+            services.TryAddSingleton<ISilo, SiloWrapper>();
+            services.TryAddFromExisting<ILocalSiloDetails, SiloInitializationParameters>();
+            services.TryAddSingleton(sp => sp.GetRequiredService<SiloInitializationParameters>().ClusterConfig);
+            services.TryAddSingleton(sp => sp.GetRequiredService<SiloInitializationParameters>().GlobalConfig);
+            services.TryAddTransient(sp => sp.GetRequiredService<SiloInitializationParameters>().NodeConfig);
+            services.TryAddSingleton<Func<NodeConfiguration>>(
```

Comment:
Thanks, created https://github.com/dotnet/orleans/issues/3312 to track it, since this is just a move of the existing registrations, and there's several other occurrences. No need to address in this particular PR just yet.